src/main/java/ui/MenuControl.java

```diff
@@ -115,15 +143,14 @@ private void onBoardSaveAs() {
 
         List<PanelInfo> panelList = panels.getCurrentPanelInfos();
 
-        if (panelList.isEmpty()) {
-            logger.info("Did not save new board");
-            return;
-        }
-
         BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
         Optional<String> response = dlg.showAndWait();
         ui.showMainStage();
-        panels.selectFirstPanel();
+        if (panelList.isEmpty()) {
+            logger.info("Saved an empty board");
+        } else {
+            panels.selectFirstPanel();
```

Comment:
What's the purpose of this line? Is this the right place to have it. The SLAP of this method can be improved.